src/main/java/pl/edu/pja/nyan/service/LessonService.java

```diff
@@ -39,12 +47,17 @@
      */
     public LessonDTO save(LessonDTO lessonDTO) {
         log.debug("Request to save Lesson : {}", lessonDTO);
-        Lesson lesson = lessonMapper.toEntity(lessonDTO);
-        lesson = lessonRepository.save(lesson);
-        if (lessonDTO.getRawTags() != null) {
-            tagService.parseTags(lessonDTO.getRawTags(), lesson);
+        Lesson lesson = lessonRepository.save(lessonMapper.toEntity(lessonDTO));
+        Set<Tag> parsedTags = tagService.parseTags(lessonDTO.getRawTags());
+        parsedTags.stream()
+            .filter(e -> e.getLessons().stream().noneMatch(l -> l.getId().equals(lesson.getId())))
+            .forEach(e -> e.addLesson(lesson));
+        Lesson lessonWithDeletedTags = deleteOldTags(lesson, parsedTags);
+        Tag lessonTag = tagService.addLessonTag(lessonWithDeletedTags);
```

Comment:
Przy utworzeniu nowej lekcji dodaje się tag o takiej samej nazwie?

Reply:
Tak, chodzi o to, że nie przypisujemy słów bezpośrednio do lekcji tylko zawsze do tagów. Każda lekcja ma swój tag, dodawany zawsze niezależnie od tego co wpiszemy. Do tego tagu przypisywane są słowa, które dodajemy do lekcji.